lesson05/circle_test.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"errors"
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const tolerance float64 = 0.00001
+
+func withTolerane(a, b, tolerance float64) bool {
+	if diff := math.Abs(a - b); diff < tolerance {
+		//	fmt.Printf("When a=%f and b =%f => Nearly same by tolerance\n", a, b)
+		return true
+	} else {
+		//	fmt.Printf("When a=%f and b=%f => Not same Even by Tolerance\n", a, b)
+		return false
+	}
+}
+func TestAreaC(t *testing.T) {
+
+	tests := []struct {
+		name    string
+		args    Circle
+		want    float64
+		wantErr error
+	}{
+		{
+			name:    "if radius is <= 0 should return error ",
+			args:    Circle{Radius: -1.0},
+			want:    0.0,
+			wantErr: errors.New("Error is happen! Radius could not be <= 0."),
+		},
+		{
+			name:    "if radius is <= 0 should return error ",
+			args:    Circle{Radius: 0.0},
+			want:    0.0,
+			wantErr: errors.New("Error is happen! Radius could not be <= 0."),
+		},
+		{
+			name:    "if radius = 1 should return Pi",
+			args:    Circle{Radius: 1.0},
+			want:    math.Pi,
+			wantErr: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := (tt.args).Area()
+
+			if err != nil {
```

Comment:
You technically don't even need to have this if statement: `assert.Equal` will fail the test if the values are not equal, and it also lets you to assert equality on nil error

Reply:
if err != nil adjusted